### DISCUSSION TITLE
CAN NOT CREATE TAG HAS A PROPERTY WHOSE TYPE IS MAP

### DISCUSSION THREAD

**Discussion by PhungQuocLuong-git** [vesoft-inc/nebula#4997] [General]
I trying to create a tag has a map-type-property:
CREATE TAG IF NOT EXISTS measurement(data map{unit: string, amount: string, family: string}, locale string);

But I have an error: 
-1004:SyntaxError: syntax error near `map'

Please help me.

**Comment by wey-gu** [comment]
Dear @PhungQuocLuong-git 

Welcome to the nebulagraph community, those composite datatype can be only supported on the fly during the query, and cannot be as prop data type for now in NebulaGraph, if you anyway would like to use it, could consider flattening them as separated props or as JSON_STRING(there is a json_extract function to parse that when we have to use).
